controller/system_backup_controller.go

```diff
@@ -547,6 +588,11 @@ func (c *SystemBackupController) cleanupRemoteSystemBackupFiles(systemBackup *lo
 
 	log := getLoggerForSystemBackup(c.logger, systemBackup.Name)
 
+	if systemBackup.Status.Version == "" {
+		// The backup store sync might not have finished
+		return nil
+	}
```

Comment:
Why this matters?

BTW, please udpate the message of the error in the following if-block.

```
if systemBackupCfg != nil {
		return errors.New("failed to check the system backup deletion, because it's being deleted")
	}

```

Reply:
> Why this matters?
> 

If the system backup exists in the backup store then we
1. Tag the SystemBackup with the volume label -> update SystemBackup.
1. The controller remote system backup info -> update SystemBackup.status.
If the SystemBackup gets deleted somewhere in between the transmission state then it might not have the version to use to validate the deletion and cause a blocking.

So for the above case, we can just continue deleting the CR, and later the backup target sync will kick back in and the user can remove it again.